src/inspector.rs

```diff
@@ -384,6 +385,12 @@ mod tests {
     fn base_mut(&mut self) -> &mut ChannelBase {
       &mut self.base
     }
+    fn base_ptr(_this: *const Self) -> *const ChannelBase
+    where
+      Self: Sized,
+    {
+      _this as *const ChannelBase
```

Comment:
This should return a raw pointer to the `base` field (analogous to `T::base()` and `T::base_mut()`), shouldn't it? That's what the code did before. Now it returns a pointer to `Self`.

Reply:
Your right because base and base_mut call base right here.
```rust   impl ChannelImpl for TestChannel {
    fn base(&self) -> &ChannelBase {
      &self.base
    }
    fn base_mut(&mut self) -> &mut ChannelBase {
      &mut self.base
    }
```
```


So since we are working with a raw pointer here something more along the lines of, this?
```rust
fn base_ptr(_this: *const Self) -> *const ChannelBase
    where
      Self: Sized,
    {
      unsafe {&(*_this).base}
    } 
```

Reply:
Yes almost, but instead of creating a reference to the base, it's better to use the [`addr_of`](https://doc.rust-lang.org/stable/std/ptr/macro.addr_of.html) macro:
```rs
fn base_ptr(_this: *const Self) -> *const ChannelBase
    where
      Self: Sized,
    {
      unsafe { addr_of!((*_this).base) }
    } 
```
This creates a `*const ChannelBase` directly without the intermediate reference

Reply:
Adjusted to follow the convention above.

Reply:
I will fix up those CLA issues tomorrow or some day or over the weekend.